File: JobTracker.Application/DTOs/ApplicationNoteDTO.cs
using JobTracker.Domain.Models.Enums;
using System.ComponentModel.DataAnnotations;

namespace JobTracker.Application.DTOs
{
    public class ApplicationNoteDTO
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "Please provide the Job Application Id.")]
        public int JobApplicationId { get; set; }

        [Required(ErrorMessage = "Please provide the Note Type.")]
        public NoteType Type { get; set; }

        [Required(ErrorMessage = "Please provide the Note Content.")]
        [MaxLength(2000, ErrorMessage = "The note content cannot exceed 2000 characters.")]
        public string Content { get; set; } = string.Empty;
    }
}

File: jobtracker-frontend/src/types/dto/application-note.dto.ts
import { NoteType } from '../enums/note-type.enum';

export interface ApplicationNoteDTO {
  id: number;
  jobApplicationId: number;
  type: NoteType;
  content: string;
}

export interface CreateApplicationNoteDTO {
  jobApplicationId: number;
  type: NoteType;
  content: string;
}

export interface UpdateApplicationNoteDTO {
  id: number;
  jobApplicationId: number;
  type: NoteType;
  content: string;
}